src/ORM/Table.php

```diff
@@ -3204,16 +3205,6 @@ public function implementedEvents(): array
             'Model.beforeRules' => 'beforeRules',
             'Model.afterRules' => 'afterRules',
         ];
-        $events = [];
-
-        foreach ($eventMap as $event => $method) {
```

Comment:
Isnt this removing the opt in part of it?

Reply:
I don't understand your question but this code is unnecessary anymore because all events listed in `$eventMap` are implemented so `$eventMap` and `$events` at the bottom are the same.

Reply:
This `foreach` needs to stay since we want the trait usage to be opt-in. I already mentioned in the other PR that always having the methods prevents once for typing the `$entity` argument with a concrete entity class.

Reply:
>  ... prevents one from typing the $entity argument with a concrete entity class.

This I do not fully understand. Could you give an example?

Reply:
> This I do not fully understand. Could you give an example?

Right now people have to do
https://github.com/dereuromark/cakephp-sandbox/blob/a95eb526074d05467553e99de8c2341abb9409c3/plugins/StateMachineSandbox/src/Model/Table/RegistrationsTable.php#L111
in order to be able to have the correct concrete object typed inside the callback.
I works fine so far for me at least (IDEHelper automatically fixes this up).

Reply:
If this trait is added to the `Table` class then in my `PostsTable`  I can't have
```php
public function beforeSave(EventInterface $event, Post $entity, ArrayObject $options): void
{
  //...
}
```

Reply:
Is phpstan also on board here?
With the overwrite?

Reply:
An example of the fatal error: https://onlinephp.io/c/f4e51

Reply:
It will throw an error if the signature is changed in the child class.

So if were to have this trait, it would have to be included in each table class separately and not `Table` (which can be done when baking).

Would have to check if static analyzers/IDEs complain if you override trait methods.

Reply:
Do you suggest to remove `use TableEventsTrait;` from the Table class?

Reply:
Yes